pilot/pkg/model/push_context.go

```diff
@@ -864,7 +864,15 @@ func addHostsFromMeshConfig(ps *PushContext, hosts sets.String) {
 // servicesExportedToNamespace returns the list of services that are visible to a namespace.
 // namespace "" indicates all namespaces
 func (ps *PushContext) servicesExportedToNamespace(ns string) []*Service {
-	out := make([]*Service, 0)
+	var capacity int
+	if ns == NamespaceAll {
```

Comment:
may be do it in the corresponding if else below so that it is more readable

Reply:
updated, thanks